src/java/org/apache/cassandra/FilterExperiment.java

```diff
@@ -44,57 +46,101 @@
+    private static final Counter failuresForEmitting =
+            CassandraMetricsRegistry.Metrics.counter(names.createMetricName("FailuresForEmitting"));
+    private static final Counter indeterminateForEmitting =
+            CassandraMetricsRegistry.Metrics.counter(names.createMetricName("IndeterminateForEmitting"));
 
     public static ColumnFamily execute(
             Function<FilterExperiment, ColumnFamily> function,
             Function<FilterExperiment, ColumnFamily> fallback) {
         if (!shouldRunExperiment()) {
             return function.apply(USE_LEGACY);
         }
-        ColumnFamily legacyResult = time(() -> function.apply(USE_LEGACY), legacyTimer);
-        try {
-            ColumnFamily optimizedResult = time(() -> function.apply(USE_OPTIMIZED), optimizedTimer);
-            ComparisonResult initialComparison = areEqual(legacyResult, optimizedResult);
-            if (initialComparison.isEqual()) {
-                successes.inc();
-            } else if (!areTrulyEqual(legacyResult, function.apply(USE_LEGACY))) {
-                // This case means that the data was modified between getting the legacy and optimized result. As a result, this experiment is indeterminate.
-                indeterminate.inc();
-            } else if ((legacyResult.metadata().getGcGraceSeconds() == 0
-                           && areEqual(fallback.apply(USE_LEGACY), fallback.apply(USE_OPTIMIZED)).isEqual())) {
-                indeterminate.inc();
-                // TODO(lkjaerozhang): Give a better log message when I actually understand what this means.
-                //  The indeterminate codepath seems to have never been hit so it's probably fine to defer for now as
-                //  long as we still have signal if we do hit it.
-                log.warn("Query result changed under immediate compaction but results from 60 seconds ago are identical, result is indeterminate; Legacy: {}, Optimized: {}, Legacy metadata: {}, Optimized metadata: {}",
-                         legacyResult, optimizedResult, safeLoggableColumnFamilyMetadata("legacyMetadata", legacyResult), safeLoggableColumnFamilyMetadata("optimizedMetadata", optimizedResult));
-            } else {
-                failures.inc();
-                log.warn("Comparison failure while experimenting; Legacy: {}, Optimized: {}, Comparison method: {}, Legacy metadata: {}, Optimized metadata: {}",
-                         legacyResult, optimizedResult, SafeArg.of("comparisonMethod", initialComparison.name()), safeLoggableColumnFamilyMetadata("legacyMetadata", legacyResult), safeLoggableColumnFamilyMetadata("optimizedMetadata", optimizedResult));
+
+        List<FilterExperiment> experiments = Arrays.asList(FilterExperiment.values());
+        Collections.shuffle(experiments); // shuffle to prevent caching bias
+
+        Map<FilterExperiment, ColumnFamily> results = new HashMap<>();
+        for (FilterExperiment experiment : experiments)
```

Comment:
Are we concerned at all with harming performance running all 3 of these? Should we rather just pick one experiment to run each time?

Reply:
Did we end up testing this? If my understanding is correct, this might add 50% more latency on that code path for 1/100 queries which seems not that negligible 🤔